### DISCUSSION TITLE
Profiling resolver memory/cpu usuage.

### DISCUSSION THREAD

**Discussion by lemiesz** [MichalLytek/type-graphql#809] [Q&A]
Hey, I'm wondering if anyone has any recommendation on methods for profiling CPU/Memory usage on a TypeGQL service. 
Specifically I am seeing some high cpu usage, and am interested in finding where I could optimize my code.   

Im open to specific ways that this can be done using the TypeGQL framework, or if there are more generalized methods for profiling a NodeJS service, I would be open to that as well.

**Comment by MichalLytek** [comment]
TypeGraphQL (not TypeGQL) specific approach might be only a resolver middleware to measure computing time:
https://typegraphql.com/docs/middlewares.html#what-is-middleware

But in Node.js you can use Chrome debugging tools:
https://medium.com/@paul_irish/debugging-node-js-nightlies-with-chrome-devtools-7c4a1b95ae27
